### ISSUE TITLE
NullReferenceException in Request.cs on SlackSocketClient.Connect(..)

### ISSUE THREAD

**Issue by FourOhs** [Inumedia/SlackAPI#46]
I'm trying to get a simple "Hello, World" custom bot working with SlackAPI.  I created a new Windows console app, and installed SlackAPI through NuGet.

I already have an auth token generated by Slack, so, step 1, I figured, is to execute .Connect() and get back a client that I can work with further.  To do that, I borrowed the "Creating a SlackSocketClient instance" example from the wiki:

https://github.com/Inumedia/SlackAPI/wiki/Examples#creating-a-slacksocketclient-instance

`var client = new SlackSocketClient(CUSTOM_BOT_AUTH_TOKEN_FROM_SLACK);`
`client.Connect((connected) =>`
`{`
    `//This is called once the RTM client has emitted the RTM start command`
    `System.Console.WriteLine("The RTM client has emitted the start command");`
`}, () =>`
`{`
`//This is called once the RTM client has connected to the end point`
`System.Console.WriteLine("The RTM client has connected to the end point");`
`});`

When I call SlackSocketClient.Connect(..), as above, I get a NullReferenceException and the IDE prompts me to find Request.cs.  At that point, the Call Stack looks like this:

`- SlackAPI.dll!SlackAPI.SlackSocket.SlackSocket(SlackAPI.LoginResponse loginDetails, object routingTo, System.Action onConnected) Line 66    C#`
`- SlackAPI.dll!SlackAPI.SlackSocketClient.ConnectSocket(System.Action onSocketConnected) Line 47    C#`
`- SlackAPI.dll!SlackAPI.SlackSocketClient.Connect.AnonymousMethod__0(SlackAPI.LoginResponse s) Line 35    C#`
`> SlackAPI.dll!SlackAPI.SlackClient.Connect.AnonymousMethod__0(SlackAPI.LoginResponse loginDetails) Line 70    C#`
`- SlackAPI.dll!SlackAPI.RequestState<SlackAPI.LoginResponse>.GotResponse(System.IAsyncResult result) Line 91    C#`
`[External Code]`

I feel I must be missing something very elementary given that my attempt at a "Hello, World" was stumped right out of the gate.

**Comment by Inumedia** [comment]
Hey FourOhs,

This probably stems from differences between bot auth tokens and regular
user auth tokens. When I originally wrote this API bots didn't exist. I'll
look into this and get back to you, in the meantime, can you try with a
regular user auth token?

**Comment by FourOhs** [comment]
Thanks, Andy.  I appreciate it.  In the meantime, as you suggested, I'll
see how it goes with a regular user token.

Cheers,
~Jeff

**Comment by FourOhs** [comment]
Well, I think maybe I found the problem.  I've switched my "Hello, World"
to be based on the other Wiki example, "Programatically getting your auth
token from your email and password credentials"

https://github.com/Inumedia/SlackAPI/wiki/Examples#programatically-getting-your-auth-token-from-your-email-and-password-credentials

After calling SlackClient.StartAuth(..) calls the AuthStartResponse
callback, the authStartResponse.error property is set to "please_upgrade".
I am inferring from this that the NuGet package I installed (SlackAPI
1.0.4.2) is perhaps in need of an update to keep up with Slack's API.

Does that sound like I've got the gist of the problem?  If so, I'll dump
the NuGet installation in favour of building from source.

Related to this, the SlackClient.StartAuth() method is marked Obsolete, so
I realize I shouldn't be calling it, and should (from the warning) be using
OAuth instead, but the closest I can find to "Hello, World" examples are
those on the Wiki: SlackClient.StartAuth(), and SlackSocketClient.Connect().
